crates/iceberg-catalog/src/request_metadata.rs

```diff
@@ -45,12 +51,14 @@ impl RequestMetadata {
         }
     }
 
+    #[cfg(test)]
     #[must_use]
     pub fn new_unauthenticated() -> Self {
         Self {
             request_id: Uuid::now_v7(),
             project_id: None,
             authentication: None,
+            host: Some("localhost".to_string()),
```

Comment:
The host as we currently use it has the protocol prepended, I would change this example to match a typical value: `http://localhost:8181`.